frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/BaseTraverseAction.java

```diff
@@ -0,0 +1,133 @@
+package com.virtuslab.gitmachete.frontend.actions.traverse;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getNonHtmlString;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import java.util.Collections;
+
+import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import git4idea.branch.GitBrancher;
+import git4idea.repo.GitRepository;
+import kr.pe.kwonnam.slf4jlambda.LambdaLogger;
+import lombok.CustomLog;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UI;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.frontend.actions.base.BaseGitMacheteRepositoryReadyAction;
+import com.virtuslab.gitmachete.frontend.actions.base.IBranchNameProvider;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.InfoDialog;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.ui.api.table.BaseEnhancedGraphTable;
+import com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils;
+
+@ExtensionMethod({GitVfsUtils.class, GitMacheteBundle.class})
+@CustomLog
+public abstract class BaseTraverseAction extends BaseGitMacheteRepositoryReadyAction implements IBranchNameProvider {
+
+  @Override
+  public LambdaLogger log() {
+    return LOG;
+  }
+
+  public static final String SHOW_TRAVERSE_INFO = "git-machete.traverse.approval.show";
+
+  @Override
+  @UIEffect
+  protected void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+    val presentation = anActionEvent.getPresentation();
+    if (!presentation.isEnabledAndVisible()) {
```

Comment:
Btw should every `onUpdate` have such check by default? Ofc not sth to be done right now, but if so, this could be enforced statically

Reply:
Wow it looks that there are maaany versions of this check 😅 

<img width="823" alt="image" src="https://user-images.githubusercontent.com/3383210/209352937-ac7dccbc-6630-48d3-8bed-28d7aceb1927.png">

Reply:
https://github.com/VirtusLab/git-machete-intellij-plugin/pull/1399 then